lib/Xliff.js

```diff
@@ -816,6 +816,11 @@ Xliff.prototype.deserialize = function(xml) {
                             tu.resname = tu.source["$t"];
                         }
 
+                        if (tu.source["x-key"]) {
+                             tu.source["$t"]= tu.source["x-key"];
+                             tu.resname = tu.source["x-key"];
+                        }
+
```

Comment:
These lines should probably be moved up to line 814 instead. The lines 815-817 are the last resort/default case...

Reply:
I updated it